### PR TITLE
Bug 1908159: operator: fix error syncing cloud config with CA data but no cloud config

### PR DESCRIPTION
**Pull request by staebler**
The machine-config-operator should accept a kube-cloud-config ConfigMap on platforms that do not require a cloud.conf when the ConfigMap has no cloud.conf data but does have ca-bundle.pem data. This was the intent of https://github.com/openshift/machine-config-operator/pull/2208, but there was a bug in that code.

### CODE REVIEW

File: pkg/operator/sync_test.go [hunk 1]
```diff
@@ -1,14 +1,152 @@
...
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 )
 
-func TestIsCloudConfigRequired(t *testing.T) {
-	testInfra := configv1.Infrastructure{}
-	testInfra.Status.Platform = "None"
-	required := isCloudConfigRequired(&testInfra)
-	assert.False(t, required)
+func TestSyncCloudConfig(t *testing.T) {
```
Comment (kikisdeliveryservice) [inline pkg/operator/sync_test.go:16]: Did we lose the platform = "none" as a case?

Comment (staebler) [reply pkg/operator/sync_test.go:16]: I am happy to add a test case for platform "None". Is there something special about that platform that you would like to me test?

Comment (staebler) [reply pkg/operator/sync_test.go:16]: Add a test case for None platform with no kube-cloud-config.

**Review by kikisdeliveryservice — APPROVED** [review]
Will let @sinnykumari do the final review
